### DISCUSSION TITLE
Is it possible to create resources separately and have terraform configure?

### DISCUSSION THREAD

**Discussion by pghdm** [microsoft/PubSec-Info-Assistant#870] [Q&A]
I'd like to downsize the app services, function app to a lower pricing tier. Tried doing through variables.tf but it complains about worker threads/dynamic SKUs etc. Is it possible to create my own resources and have Terraform configure it?

**Comment by bjakems** [comment]
To a degree, yes.  Terraform must be aware about current infrastructure, so you would need to import your resources into the Terraform state.  Also, please ensure the terraform name for that resource in the TF code aligns with your created resource.  It is easier to scale up / down resources leveraging the local.env file or manually in the portal (if manually in the portal, please refresh your tfstate as it will not have these updates).  Due to memory and CPU requirements, I do not recommend scaling down from the following configurations:
(local.env entries)
export TF_VAR_functionsAppSkuSize="S2"
export TF_VAR_functionsAppSkuTier="Standard"
export TF_VAR_appServiceSkuSize="S1"
export TF_VAR_appServiceSkuTier="Standard"
export TF_VAR_enrichmentAppServiceSkuSize="S3"
export TF_VAR_enrichmentAppServiceSkuTier="Standard"
